test/parallel/test-dgram-multicast-set-interface-lo.js

```diff
@@ -0,0 +1,288 @@
+// $ socat UDP-RECVFROM:12356,ip-add-membership=224.0.0.115:127.0.0.1,fork \
+//   EXEC:hostname &
+// $ echo hi |socat STDIO \
+//   UDP4-DATAGRAM:224.0.0.115:12356,ip-multicast-if=127.0.0.1
+
+if (common.isSunOS) {
+  common.skip('SunOs is not correctly delivering to loopback multicast.');
+  return;
+}
+
+const networkInterfaces = require('os').networkInterfaces();
+const Buffer = require('buffer').Buffer;
+const fork = require('child_process').fork;
+const MULTICASTS = {
+  IPv4: ['224.0.0.115', '224.0.0.116', '224.0.0.117'],
+  IPv6: ['ff02::1:115', 'ff02::1:116', 'ff02::1:117']
+};
+const LOOPBACK = {IPv4: '127.0.0.1', IPv6: '::1'};
+const ANY = {IPv4: '0.0.0.0', IPv6: '::'};
+const FAM = 'IPv4';
+
+// Windows wont bind on multicasts so its filtering is by port.
+const PORTS = {};
+for (let i = 0; i < MULTICASTS[FAM].length; i++) {
+  PORTS[MULTICASTS[FAM][i]] = common.PORT + (common.isWindows ? i : 0);
+}
+
+const UDP = {IPv4: 'udp4', IPv6: 'udp6'};
+
+const TIMEOUT = common.platformTimeout(5000);
+const NOW = Date.now();
+const TMPL = (tail) => `${NOW} - ${tail}`;
+
+// Take the first non-internal interface as the other interface to isolate
+// from loopback. Ideally, this should check for whether or not this interface
+// and the loopback have the MULTICAST flag.
+const interfaceAddress = ((networkInterfaces) => {
+  for (const name in networkInterfaces) {
+    for (const localInterface of networkInterfaces[name]) {
+      if (!localInterface.internal && localInterface.family === FAM) {
+        let interfaceAddress = localInterface.address;
+        // On Windows, IPv6 would need: `%${localInterface.scopeid}`
+        if (FAM === 'IPv6')
+          interfaceAddress += `${interfaceAddress}%${name}`;
+        return interfaceAddress;
+      }}}})(networkInterfaces);
+
+assert.ok(interfaceAddress);
+
+const messages = [
+  {tail: 'First message to send', mcast: MULTICASTS[FAM][0], rcv: true},
+  {tail: 'Second message to send', mcast: MULTICASTS[FAM][0], rcv: true},
+  {tail: 'Third message to send', mcast: MULTICASTS[FAM][1], rcv: true,
+   newAddr: interfaceAddress},
+  {tail: 'Fourth message to send', mcast: MULTICASTS[FAM][2]},
+  {tail: 'Fifth message to send', mcast: MULTICASTS[FAM][1], rcv: true},
+  {tail: 'Sixth message to send', mcast: MULTICASTS[FAM][2], rcv: true,
+   newAddr: LOOPBACK[FAM]}
+];
+
+
+if (process.argv[2] !== 'child') {
+  const IFACES = [ANY[FAM], interfaceAddress, LOOPBACK[FAM]];
+  const workers = {};
+  const listeners = MULTICASTS[FAM].length * 2;
+  let listening = 0;
+  let dead = 0;
+  let i = 0;
+  let done = 0;
+  let timer = null;
+  // Exit the test if it doesn't succeed within the TIMEOUT.
+  timer = setTimeout(function() {
+    console.error('[PARENT] Responses were not received within %d ms.',
+                  TIMEOUT);
+    console.error('[PARENT] Fail');
+
+    killChildren(workers);
+
+    process.exit(1);
+  }, TIMEOUT);
+
+  // Launch the child processes.
```

Comment:
I have same question, it seems completely unnecessary. If you do fork children, the children should exit if the parent does to prevent leaking: `process.on('disconnect', () => process.exit(1))`